cli/cmd/aeon.go

```diff
@@ -80,7 +85,84 @@ func NewAeonCmd() *cobra.Command {
 }
 
 func aeonConnectValidateArgs(cmd *cobra.Command, args []string) error {
-	connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(args[0])
+
+	if len(args) == 1 && util.IsURL(args[0]) {
+		connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(args[0])
+	} else if len(args) == 2 && util.IsRegularFile(args[0]) {
+		var (
+			configPath = args[0]
+			instance   = args[1]
+		)
+
+		f, err := os.ReadFile(configPath)
+		if err != nil {
+			return err
+		}
+
+		pb := cluster.NewYamlCollector(f)
+		config, err := pb.Collect()
+		if err != nil {
+			return err
+		}
+
+		clusterConfig, err := cluster.MakeClusterConfig(config)
+		if err != nil {
+			return err
+		}
+
+		result := cluster.Instantiate(clusterConfig, instance)
+
+		// Get SSL connection.
+		dataSsl := []string{"roles_cfg", "aeon.grpc", "advertise"}
+		data, err := result.Get(dataSsl)
+		if err != nil {
+			return err
+		}
+
+		advertise, ok := data.(map[any]any)
+		if !ok {
+			return errors.New("data it is impossible to result in a map")
+		}
+
+		uri, ok := advertise["uri"]
+		if !ok {
+			return errors.New("filed uri not found")
+		}
+
+		re := regexp.MustCompile("^https?://")
+		cleanedURL := re.ReplaceAllString(uri.(string), "")
+
+		connectCtx.Network, connectCtx.Address = libconnect.ParseBaseURI(cleanedURL)
+
+		params, ok := advertise["params"]
+		if !ok {
+			return errors.New("filed params not found")
+		}
+
+		paramsMap, ok := params.(map[any]any)
+		if !ok {
+			return errors.New("params it is impossible to result in a map")
+		}
+
+		if transport, ok := paramsMap["transport"]; ok {
+			if (transport.(string) != "ssl") && (transport.(string) != "plain") {
+				return errors.New("transport must be ssl or plain")
+			}
+
+			if transport.(string) == "ssl" {
+				connectCtx.Transport = aeoncmd.TransportSsl
+
+				err = mapstructure.Decode(paramsMap, &connectCtx.Ssl)
+				if err != nil {
+					return err
+				}
+
+			} else {
+				connectCtx.Transport = aeoncmd.TransportPlain
+			}
+		}
```

Comment:
Nit, I mean to parse a whole structure with mapstructure:
```
                params := struct{
                    Uri   string
                    Params struct{
                        Transport string
                        KeyFile string `mapstructure:"ssl_key_file"`
                        CertFile string `mapstructure:"ssl_cert_file"`
                        CaFile string `mapstructure:"ssl_ca_file"`
                    }
                }
                if err := mapstructure.Decode(advertise, &params); err != nil {
                    return fmt.Errorf("failed to parse roles_cfg.aeon.grpc.advertise: %w", err)
                }

                // Code to check/use parsed values.
```

Reply:
I think we also need to be careful with flags here. For example, if a config have ssl_key_file, but we have a flag value for that - we need to prefer a flag value.

Reply:
done